src/main/java/li/cil/tis3d/common/module/ExecutionModule.java

```diff
@@ -64,12 +64,15 @@
         WAIT
     }
 
-    private static final Identifier[] STATE_LOCATIONS = new Identifier[]{
-        Textures.LOCATION_OVERLAY_MODULE_EXECUTION_IDLE,
-        Textures.LOCATION_OVERLAY_MODULE_EXECUTION_ERROR,
-        Textures.LOCATION_OVERLAY_MODULE_EXECUTION_RUNNING,
-        Textures.LOCATION_OVERLAY_MODULE_EXECUTION_WAITING
-    };
+    @Environment(EnvType.CLIENT)
+    private static final class RenderData {
```

Comment:
Feel free to only answer this only here or on the 1.16 pr comment: does adding the client-only annotation directly to the field not world? It appears to work in the `DisplayModule`; does this being a static field make a difference?

Reply:
This was a long time ago, but I remember trying a lot of ways and settling on the one which had the least amount of changes. I'll try again.